Addons/tsgExoticQuadrature.hpp

```diff
@@ -178,28 +197,77 @@ inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double sh
                                     std::move(weights_cache), description);
 }
 
-// Basic function overloads.
-
-// If only the weight function is provided, the reference points and weights are generated by the GL rule and the weight_fn values
-// are generated by applying the weight function to these reference points.
+/*
+ * \ingroup TasmanianAddons
+ * \brief Constructs a CustomTabulated object containing the exotic quadrature nodes and quadrature weights for a given level \b n,
+ * scalar \b shift, a reference TasmanianSparseGrid \b grid containing loaded values of the weight function, a string \b description,
+ * and an optional bool \b is_symmetric that should be set to true if the weight function is symmetric.
+ *
+ * It is assumed that grid has the following properties:
+ *     (i)   grid.getPoints() returns the quadrature reference points used to compute the entries of the Jacobi matrix.
+ *     (ii)  grid.getQuadratureWeights() returns the quadrature reference weights corresponding to grid.getPoints().
+ *     (iii) grid.getLoadedValues() returns the values of the weight function at grid.getPoints().
+ */
+inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift, TasGrid::TasmanianSparseGrid grid,
```

Comment:
Are you sure that you need to make a copy of the sparse grid as opposed to taking `const &`.

Reply:
Good point. I changed the function signature.